internal/infrastructure/db/service_test.go

```diff
@@ -1285,6 +1286,101 @@ func testConvictionRepository(t *testing.T, svc ports.RepoManager) {
 	})
 }
 
+func testFeeRepository(t *testing.T, svc ports.RepoManager) {
+	t.Run("test_fee_repository", func(t *testing.T) {
+		ctx := context.Background()
+		repo := svc.Fees()
+
+		// fees should be initialized to 0.0
+		currentFees, err := repo.GetIntentFees(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, currentFees)
+		require.Equal(t, "0.0", currentFees.OnchainInputFee)
+		require.Equal(t, "0.0", currentFees.OffchainInputFee)
+		require.Equal(t, "0.0", currentFees.OnchainOutputFee)
+		require.Equal(t, "0.0", currentFees.OffchainOutputFee)
+
+		newFees := domain.IntentFees{
+			OnchainInputFee:   "0.25",
+			OffchainInputFee:  "0.30",
+			OnchainOutputFee:  "0.35",
+			OffchainOutputFee: "0.40",
+		}
+
+		// sqlite and postgres use millisecond precision for created_at so we need to
+		// wait to ensure the updated_at is different.
+		// set the new fees
+		time.Sleep(10 * time.Millisecond)
+		err = repo.UpdateIntentFees(ctx, newFees)
+		require.NoError(t, err)
+
+		updatedFees, err := repo.GetIntentFees(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, updatedFees)
+		require.Equal(t, newFees.OnchainInputFee, updatedFees.OnchainInputFee)
+		require.Equal(t, newFees.OffchainInputFee, updatedFees.OffchainInputFee)
+		require.Equal(t, newFees.OnchainOutputFee, updatedFees.OnchainOutputFee)
+		require.Equal(t, newFees.OffchainOutputFee, updatedFees.OffchainOutputFee)
+		time.Sleep(10 * time.Millisecond)
+		// zero out the fees
+		err = repo.ClearIntentFees(ctx)
+		require.NoError(t, err)
+
+		clearedFees, err := repo.GetIntentFees(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, clearedFees)
+		require.Equal(t, "0.0", clearedFees.OnchainInputFee)
+		require.Equal(t, "0.0", clearedFees.OffchainInputFee)
+		require.Equal(t, "0.0", clearedFees.OnchainOutputFee)
+		require.Equal(t, "0.0", clearedFees.OffchainOutputFee)
+
+		// set the fees back to newFees
+		time.Sleep(10 * time.Millisecond)
+		err = repo.UpdateIntentFees(ctx, newFees)
+		require.NoError(t, err)
+
+		updatedFees, err = repo.GetIntentFees(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, updatedFees)
+		require.Equal(t, newFees.OnchainInputFee, updatedFees.OnchainInputFee)
+		require.Equal(t, newFees.OffchainInputFee, updatedFees.OffchainInputFee)
+		require.Equal(t, newFees.OnchainOutputFee, updatedFees.OnchainOutputFee)
+		require.Equal(t, newFees.OffchainOutputFee, updatedFees.OffchainOutputFee)
```

Comment:
this is basically repeating the prev code, that we already tested. What if, instead, here we check sometihng similar to the case below?

with an "empty" db we should be able to update the fees partially too

Reply:
yeah, my idea here is to not re-do exactly the things done 'till L1314, rather we can take the occasion to test smtn different, ie a partial update of the fees with empty db (where empty db means all values in db are "0.0"), and we keep also what's from L1351 to test partial update also on non-empty db

Reply:
gotcha so the test flow would be:

empty db -> partial update -> full update -> ClearIntentFees

is this correct?